### PR TITLE
gdal: 3.6.4 -> 3.7.0

### LINKED ISSUE #231955: Update request: gdal 3.6.4 → 3.7
- Package name: gdal
- Latest released version: 3.7.0 ([release notes](https://github.com/OSGeo/gdal/blob/v3.7.0/NEWS.md))
- Current version on the unstable channel: 3.6.4
- Current version on the stable/release channel: 3.6.2
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@dotlambda @MarcWeber

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by kidanger**
###### Description of changes

Diff: https://github.com/OSGeo/gdal/compare/v3.6.4...v3.7.0

Changelog: https://github.com/OSGeo/gdal/blob/v3.7.0/NEWS.md

This PR contains one patch over the draft PR #232222. Closes #231955.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kidanger** [comment]
@ofborg build gdal gdal.passthru.tests

**Comment by imincik** [comment]
Hi @kidanger. Thank you very much for this update. I am just building this PR and will test.

If you are interested in using or maintaining Geospatial software, please consider joining [Nix Geospatial team](https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix). We are working on adding GDAL to the list of software under under our maintenance.

**Review by fabianhjr — APPROVED** [review]
Changes LGTM, builds with postgis (postgres_15) on NixOS

**Review by imincik — APPROVED** [review]
Looks good to me. I tested with QGIS loading some vector data.

**Comment by imincik** [comment]
Manually tested running following commands:

```
./result/bin/gdalinfo --formats
```

```
./result/bin/ogrinfo --formats
```

```
./result/bin/ogr2ogr -f GeoJSON /vsistdout/  ../../../osgeo/qgis/code/tests/testdata/points.shp
```

Output looks good.

**Comment by imincik** [comment]
I am trying to run `nixpkgs-review  pr --post-result 239458` on my AWS machine now.

**Comment by imincik** [comment]
Unfortunately, nixpkgs-review is failing on errors like this (which does't relate to this PR). I am not sure what's wrong.

```
CONFLICT (content): Merge conflict in pkgs/top-level/python-packages.nix
Auto-merging pkgs/top-level/release.nix
CONFLICT (content): Merge conflict in pkgs/top-level/release.nix
Auto-merging pkgs/top-level/ruby-packages.nix
CONFLICT (content): Merge conflict in pkgs/top-level/ruby-packages.nix
Automatic merge failed; fix conflicts and then commit the result.
https://github.com/NixOS/nixpkgs/pull/239458 failed to build
```

File: pkgs/development/libraries/gdal/default.nix [hunk 1]
```diff
@@ -88,6 +89,7 @@ stdenv.mkDerivation rec {
     "-DGEOTIFF_LIBRARY_RELEASE=${lib.getLib libgeotiff}/lib/libgeotiff${stdenv.hostPlatform.extensions.sharedLibrary}"
     "-DMYSQL_INCLUDE_DIR=${lib.getDev libmysqlclient}/include/mysql"
     "-DMYSQL_LIBRARY=${lib.getLib libmysqlclient}/lib/${lib.optionalString (libmysqlclient.pname != "mysql") "mysql/"}libmysqlclient${stdenv.hostPlatform.extensions.sharedLibrary}"
+    "-DBUILD_TESTING=ON"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gdal/default.nix:92]: This should depend on doCheck.

Comment (imincik) [reply pkgs/development/libraries/gdal/default.nix:92]: @SuperSandro2000 , do you mean doing something like

```
++ lib.optionals doCheck
```

?

Comment (SuperSandro2000) [reply pkgs/development/libraries/gdal/default.nix:92]: yep

Comment (imincik) [reply pkgs/development/libraries/gdal/default.nix:92]: @kidanger , are you able to update your PR with this change ? Thanks.

Comment (kidanger) [reply pkgs/development/libraries/gdal/default.nix:93]: Done, thanks!

Comment (imincik) [reply pkgs/development/libraries/gdal/default.nix:93]: Sorry, we should check `doInstallCheck` instead.

This 

```
  ] ++ lib.optionals doInstallCheck [
```

works.

**Comment by imincik** [comment]
GDAL 3.7.1 is out now.

**Comment by SuperSandro2000** [comment]
we can create a new PR for that otherwise we delay the merge of this one further

**Comment by imincik** [comment]
Thanks everybody involved in this PR :) . I'll create PR for 3.7.1.

**Comment by imincik** [comment]
Here we go with 3.7.1 + package tests - https://github.com/NixOS/nixpkgs/pull/243422

Please review. Thanks.

**Pull request merged**